### PR TITLE
ref(Jira Server): Replace usage of createmeta

### LINKED ISSUE #36357: createmeta REST endpoint removed in JIRA 9
### Environment

self-hosted (https://develop.sentry.dev/self-hosted/)

### Version

22.6.0

### Steps to Reproduce

1. Install JIRA 9
2. Connect Sentry to that JIRA 9 following [these instructions](https://docs.sentry.io/product/integrations/issue-tracking/jira/)
3. Notice 400s whenever `createmeta` endpoint is being called

Essentially, this line is no longer valid: https://github.com/getsentry/sentry/blob/master/src/sentry/integrations/jira/client.py#L79

Here are docs from JIRA side mentioning `createmeta` deprecation: https://confluence.atlassian.com/jiracore/createmeta-rest-endpoint-to-be-removed-975040986.html

Anything higher than JIRA 8.4 is going to hit this problem.

### Expected Result

Working integration like it was with the prior <9 JIRA versions.

### Actual Result

Most (if not all) endpoints in the `JiraApiClient` look OK, so it's the `/rest/api/2/issue/createmeta` in particular that is causing the problems.

Screenshot 2022-07-05 at 12 02 04

### PR DESCRIPTION
**Pull request by ceorourke**
Jira Server has deprecated the [createmeta](https://confluence.atlassian.com/jiracore/createmeta-rest-endpoint-to-be-removed-975040986.html) endpoint in v9.0^ that we use to populate the issue creation modal and to create the issue. This PR removes usages of it and replaces it with the recommendations from Atlassian.

This fixes https://github.com/getsentry/sentry/issues/36357

### CODE REVIEW

File: fixtures/integrations/jira/stub_client.py [hunk 1]
```diff
@@ -11,6 +11,18 @@ def get_create_meta_for_project(self, project):
...
+    def get_issue_fields(self, project_id, issue_type_id):
+        return self._get_stub_data("issue_fields_response.json")
+
+    def get_issue_types(self, project_id):
+        return self._get_stub_data("issue_types_response.json")
+
+    def get_priorities(self):
+        return self._get_stub_data("priorities_response.json")
+
+    def get_versions(self, project_id):
```
Comment (ceorourke) [inline fixtures/integrations/jira/stub_client.py:23]: It's a little concerning to me that I just added stubs for getting the versions and priorities, I think the Jira Cloud tests are missing coverage.

File: tests/sentry/integrations/jira_server/test_integration.py [hunk 2]
```diff
@@ -55,22 +58,22 @@ def test_get_create_issue_config(self):
             assert self.installation.get_create_issue_config(group, self.user) == [
                 {
                     "name": "project",
-                    "default": "10000",
-                    "updatesForm": True,
-                    "choices": [("10000", "EX"), ("10001", "ABC")],
                     "label": "Jira Project",
+                    "choices": [("10000", "EX"), ("10001", "ABC")],
```
Comment (ceorourke) [inline tests/sentry/integrations/jira_server/test_integration.py:62]: The diff in this test is from the new response JSON

File: tests/sentry/integrations/jira_server/test_integration.py [hunk 3]
```diff
@@ -228,13 +292,19 @@ def test_get_create_issue_config_with_ignored_fields(self):
                 "title",
                 "description",
                 "issuetype",
+                "priority",
```
Comment (ceorourke) [inline tests/sentry/integrations/jira_server/test_integration.py:295]: Same here, updated response JSON

File: src/sentry/integrations/jira_server/integration.py [hunk 4]
```diff
@@ -659,86 +659,24 @@ def build_dynamic_field(self, field_meta, group=None):
...
-                extra={"organization_id": self.organization_id, "jira_project": project_id},
-            )
-            raise IntegrationError(
-                "Jira returned: Unauthorized. " "Please check your configuration settings."
-            )
+            jira_projects = client.get_projects_list()
         except ApiError as e:
             logger.info(
-                "jira.fetch-issue-create-meta.error",
+                "jira_server.get-create-issue-config.no-projects",
```
Comment (leeandher) [inline src/sentry/integrations/jira_server/integration.py:669]: I think this log line should only reference the method that calls it directly, not the parent one. 

e.g. `jira_server.get_projects.error` 

If we want to log `get-create-issue-config` it's probably best to do so in that method itself :)

And you can probably throw in an extra log on the next line that is for `no-projects`

e.g.
```
if len(jira_projects) == 0:
    logger.info(`jira_server.get_projects.no_projects", ...)
```

File: src/sentry/integrations/jira_server/integration.py [hunk 5]
```diff
@@ -766,47 +705,45 @@ def get_create_issue_config(self, group, user, **kwargs):
...
-                extra={
-                    "integration_id": self.model.id,
-                    "organization_id": self.organization_id,
-                    "error": str(e),
-                },
-            )
-            raise IntegrationError(
-                "Could not fetch project list from Jira. Ensure that Jira is"
-                " available and your account is still active."
+                "jira_server.fetch-issue-create-meta.unauthorized",
```
Comment (leeandher) [inline src/sentry/integrations/jira_server/integration.py:733]: Same with these logs!

**Review by leeandher — APPROVED** [review]
Looks good! One small comment, and it's up to you, but do you think this is worth testing with a feature flag first prior to merging? It would change the way existing apps immediately, but if it's low risk, then feel free to ignore  🚀 🚢

File: src/sentry/integrations/jira_server/integration.py (hunk 4 above)
Comment (ceorourke) [reply src/sentry/integrations/jira_server/integration.py:669]: Good catch, idk what I was thinking when I did these lol

**Comment by ceorourke** [comment]
> Looks good! One small comment, and it's up to you, but do you think this is worth testing with a feature flag first prior to merging? It would change the way existing apps immediately, but if it's low risk, then feel free to ignore 🚀 🚢

I think it's low risk (and high effort to change everything to be feature flaggable), I tested it locally with our Jira Server (version below the breaking change one) and if the Atlassian docs are to be trusted, it will work with the higher versions too.

**Pull request merged**